source/src/main/kotlin/org/neo4j/connectors/kafka/source/SourceConfiguration.kt

```diff
@@ -253,28 +254,34 @@ class SourceConfiguration(originals: Map<*, *>) :
   private fun mapMetadata(
       configEntry: CdcPatternConfigItem,
       nonPositionalConfigMode: MutableMap<String, Boolean>,
-      configMap: MutableMap<String, MutableList<Pattern>>
+      configMap: MutableMap<String, MutableList<Pattern>>,
+      patternTxMetadataMap: MutableMap<Pattern, MutableMap<String, Any>>
   ) {
     val (index, patterns) = retrieveIndexAndPattern(configEntry, nonPositionalConfigMode, configMap)
-    val keyValue = configEntry.metadata!!
-    var value = configEntry.value
+    val metadataKey = configEntry.metadata!!
+    val value = configEntry.value
     val pattern = patterns[index]
-    if (keyValue.startsWith(EntitySelector.METADATA_KEY_TX_METADATA + '.')) {
-      value =
-          mapOf(
-              keyValue.removePrefix(EntitySelector.METADATA_KEY_TX_METADATA + '.') to value,
-          )
-      val metadata =
-          mapOf(
-              EntitySelector.METADATA_KEY_TX_METADATA to value,
-          )
-      pattern.withMetadata(metadata)
+    if (metadataKey.startsWith("$METADATA_KEY_TX_METADATA.")) {
+      val txMetadataKey = metadataKey.removePrefix("$METADATA_KEY_TX_METADATA.")
+
+      var txMetadata = patternTxMetadataMap[pattern]
+      if (txMetadata == null) {
+        txMetadata = mutableMapOf(txMetadataKey to value)
+      } else {
+        txMetadata[txMetadataKey] = value
+      }
```

Comment:
could `getOrPut` or similar methods on map work here?

Reply:
`getOrPut` works very well for this case 👍